pkg/event/target/mqtt.go

```diff
@@ -82,75 +79,49 @@ type MQTTTarget struct {
 	args   MQTTArgs
 	client mqtt.Client
 	store  Store
-	reconn bool
 }
 
 // ID - returns target ID.
 func (target *MQTTTarget) ID() event.TargetID {
 	return target.id
 }
 
-// Reads persisted events from the store and re-plays.
-func (target *MQTTTarget) retry() {
-	target.reconn = true
-	events := target.store.ListAll()
-	for len(events) != 0 {
-		for _, key := range events {
-			event, eErr := target.store.Get(key)
-			if eErr != nil {
-				continue
-			}
-			for !target.client.IsConnectionOpen() {
-				time.Sleep(retryInterval * time.Second)
-			}
-			// The connection is open.
-			if err := target.send(event); err != nil {
-				continue
-			}
-			// Delete after a successful publish.
-			target.store.Del(key)
-		}
-		events = target.store.ListAll()
+// Send - sends event to MQTT.
+func (target *MQTTTarget) Send(eventKey string) (bool, error) {
```

Comment:
Why send requires `(bool, error)` as return type.  I think `error` is good enough.  You may use typed error for specifics.